accounts/views.py

```diff
@@ -21,33 +24,40 @@ def register(request):
     if request.method == "POST":
         form = RegistrationForm(request.POST)
         if form.is_valid():
-            first_name = form.cleaned_data['first_name']
-            last_name = form.cleaned_data['last_name']
-            email = form.cleaned_data['email']
-            phone_number = form.cleaned_data['phone_number']
-            password = form.cleaned_data['password']
-            username = email.split('@')[0]
-
-            user = Account.objects.create_user(
-                                first_name=first_name,
-                                last_name=last_name,
-                                email=email,
-                                password=password,
-                                username=username
-                                )
-            user.phone_number = phone_number
-            user.save()
-
-            current_site = get_current_site(request)
-            mail_subject = "Please activate your account!"
-            message = render_to_string('accounts/account_verification_email.html', {
-                'user': user,
-                'domain': current_site,
-                'uid': urlsafe_base64_encode(force_bytes(user.id)),
-                'token': default_token_generator.make_token(user),
-            })
-            send_email = EmailMessage(mail_subject, message, to=[email])
-            send_email.send()
+            try:
+                with transaction.atomic():
+                    first_name = form.cleaned_data['first_name']
+                    last_name = form.cleaned_data['last_name']
+                    email = form.cleaned_data['email']
+                    phone_number = form.cleaned_data['phone_number']
+                    password = form.cleaned_data['password']
+                    username = email.split('@')[0]
+
+                    user = Account.objects.create_user(
+                                        first_name=first_name,
+                                        last_name=last_name,
+                                        email=email,
+                                        password=password,
+                                        username=username
+                                        )
+                    user.phone_number = phone_number
+                    user.save()
+
+                    current_site = get_current_site(request)
+                    mail_subject = "Please activate your account!"
+                    message = render_to_string('accounts/account_verification_email.html', {
+                        'user': user,
+                        'domain': current_site,
+                        'uid': urlsafe_base64_encode(force_bytes(user.id)),
+                        'token': default_token_generator.make_token(user),
+                    })
+                    send_email = EmailMessage(mail_subject, message, to=[email])
+                    send_email.send()
```

Comment:
and for this part, you should create a seperate function which accepts two params `user` and `host_site`:

```
def send_account_verification_email(user: Account, host_site: str):
    pass


site = get_host_site(request)
send_account_verification_email(user=user, host_site=site)
```

and rename `get_current_site` to `get_host_site`

Reply:
get_current_site is built in function i imported it from django shurtcuts